compiler/passes/ResolveScope.cpp

```diff
@@ -851,6 +852,34 @@ Symbol* ResolveScope::lookupPublicImports(const char* name) const {
   return retval;
 }
 
+Symbol* ResolveScope::lookupPublicUnqualAccessSyms(const char* name,
+                                                   ModuleSymbol*& modArg) const {
+  UseImportList useImportList = mUseImportList;
+
+  for_vector_allowing_0s(VisibilityStmt, visStmt, useImportList) {
+    if (ImportStmt *is = toImportStmt(visStmt)) {
+      if (!is->isPrivate) {
+        for_vector(const char, unqualName, is->unqualified) {
+          if (astr(unqualName) == astr(name)) {
+            if (SymExpr *se = toSymExpr(is->src)) {
+              if (ModuleSymbol *ms = toModuleSymbol(se->symbol())) {
+                ResolveScope *scope = ResolveScope::getScopeFor(ms->block);
+                Symbol *retval = scope->lookupNameLocally(name);
+                if (retval) {
+                  modArg = ms;
+                  return retval;
```

Comment:
I have some vague concerns about this being too permissive.  One situation that comes to mind:
An import statement further in the chain mentions the symbol, but the closer import doesn't (so it shouldn't be available).  E.g.
```chapel
module A {
   var x: int;
}
module B {
  import A.x;

  var y: bool;
}
module C {
  import B.y;
}
module User {
  use C;
  proc main() {
    writeln(C.x); // Shouldn't work
    // we do know about C, and B does pretend it has an x, but C doesn't bring that in
  }
}
```
I had another case, but in writing this one, I forgot it.  Will let you know if I remember it

Reply:
I think the other case was more "but this situation should still work if we need to fix the other one", basically a deep chain to enable a symbol

Reply:
The particular code sample you have here seems to work, but I'll dig deeper.

Reply:
I should have said "... seems to work in the sense that it does what is expected and doesn't work" :)

Reply:
Added this exact snippet and a similar one as test.